src/controller/JCMCommissioner.cpp

```diff
@@ -0,0 +1,430 @@
+                            return CHIP_ERROR_CERT_NOT_FOUND;
+                        }
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator NOC and ICAC");
+                        break;
+                    }
+                }
+                return CHIP_NO_ERROR;
+            }
+            default:
+                return CHIP_NO_ERROR;
+        }
+
+        return CHIP_NO_ERROR;
+    });
+
+    return err;
+}
+
+CHIP_ERROR JCMDeviceCommissioner::ParseTrustedRoot(ReadCommissioningInfo & info)
+{
+    auto attributeCache = info.attributes;
+
+    CHIP_ERROR err = attributeCache->ForEachAttribute(OperationalCredentials::Id, [this, &attributeCache](const ConcreteAttributePath & path) {
+        using namespace chip::app::Clusters::OperationalCredentials::Attributes;
+        bool foundMatchingRcac = false;
+
+        switch (path.mAttributeId)
+        {
+            case TrustedRootCertificates::Id: {
+                TrustedRootCertificates::TypeInfo::DecodableType trustedCAs;
+                ReturnErrorOnFailure(attributeCache->Get<TrustedRootCertificates::TypeInfo>(path, trustedCAs));
+
+                   auto iter = trustedCAs.begin();
+                   while (iter.Next())
+                   {
+                       auto & trustedCA = iter.GetValue();
+                       Credentials::P256PublicKeySpan trustedCAPublicKeySpan;
+
+                       ReturnErrorOnFailure(Credentials::ExtractPublicKeyFromChipCert(trustedCA, trustedCAPublicKeySpan));
+                       Crypto::P256PublicKey trustedCAPublicKey{ trustedCAPublicKeySpan };
+
+                       if (mInfo.rootKeySpan.size() != Crypto::kP256_PublicKey_Length)
+                       {
+                           ChipLogError(Controller, "JCM: DeviceCommissioner::ParseJFAdministratorInfo - fabric root key size mismatch");
+                           return CHIP_ERROR_KEY_NOT_FOUND;
+                       }
+
+                       Credentials::P256PublicKeySpan rootPubKeySpan(mInfo.rootKeySpan.data());
+                       Crypto::P256PublicKey fabricTableRootPublicKey{ rootPubKeySpan };
+
+                       if (trustedCAPublicKey.Matches(fabricTableRootPublicKey) && trustedCA.size())
+                       {
+                            ReturnErrorOnFailure(AllocateMemoryAndCopySpan(mInfo.adminRCAC, trustedCA));
+                           
+                            foundMatchingRcac = true;
+                            break;
+                       }
+                   }
+
+                   if (!foundMatchingRcac)
+                   {
+                       ChipLogError(Controller, "JCM: Cannot find a matching RCAC!");
+                       return CHIP_ERROR_CERT_NOT_FOUND;
+                   }
+                   return CHIP_NO_ERROR;
+               }
+               default:
+                   return CHIP_NO_ERROR;
+           }
+           return CHIP_NO_ERROR;
```

Comment:
this return CHIP_NO_ERROR even if no valid attribute if found. please also reuse the validity checks from the existing code.